crates/cext/src/circuit.rs

```diff
@@ -605,3 +605,65 @@ pub unsafe extern "C" fn qk_circuit_to_python(circuit: *mut CircuitData) -> *mut
             .into_ptr()
     }
 }
+
+/// @ingroup QkCircuit
+///
+/// C compatible enum for DelayUnit
+#[repr(C)]
+pub enum QkDelayUnit {
+    S,
+    MS,
+    US,
+    NS,
+    PS,
+    DT,
+}
```

Comment:
A few notes on this:
* I think we can make this `repr(u8)` since we don't have a lot of values
* it's good practice to fix the values here, otherwise the order of enum variants determines their values, which can be error-prone since these are just integers C-side
* I think we should write these docs for people that only use C as well. So e.g. "C compatible" is clear -- they will read this in the C API docs of Qiskit 🙂 
* It might be nice to explain the enum variants

```suggestion
/// @ingroup QkCircuit
///
/// Units for circuit delays.
#[repr(u8)]
pub enum QkDelayUnit {
    /// Seconds.
    S = 0,
    /// Milliseconds.
    MS = 1,
    /// Microseconds.
    US = 2,
    /// Nanoseconds.
    NS = 3,
    /// Picoseconds.
    PS = 4,
    /// Device-native time unit ``dt``.
    DT = 5,
}
```